src/main/java/stsjorbsmod/patches/NeowEventPatch.java

```diff
@@ -22,14 +36,38 @@ public static void Prefix(NeowEvent __this, boolean isDone) {
     public static class NeowEvent_ButtonEffect {
         public static ExprEditor Instrument() {
             return new ExprEditor() {
-                public void edit(FieldAccess methodCall) throws CannotCompileException {
-                    if (methodCall.getClassName().equals(NeowEvent.class.getName()) && methodCall.getFieldName().equals("waitingToSave")) {
-                        methodCall.replace(String.format(
+                public void edit(FieldAccess fa) throws CannotCompileException {
+                    if (fa.getClassName().equals(NeowEvent.class.getName()) && fa.getFieldName().equals("waitingToSave")) {
+                        fa.replace(String.format(
                                 "{ %1$s.doAtStartOfAct(); $_ = $proceed($$); }",
                                 AtStartOfActRelicSubscriber.class.getName()));
                     }
                 }
             };
         }
     }
+
+    @SpirePatch(clz = NeowEvent.class, method = "buttonEffect")
+    public static class NeowEvent_Cull_AddCards {
+        public static ExprEditor Instrument() {
+            return new ExprEditor() {
+                public void edit(MethodCall methodCall) throws CannotCompileException {
+                    if (methodCall.getClassName().equals(NeowEvent.class.getName()) &&
+                            (methodCall.getMethodName().equals("blessing") || methodCall.getMethodName().equals("miniBlessing"))) {
+                        methodCall.replace(String.format(
+                                "{ %1$s.addDeckOfTrialsCardsIfNecessary(); $_ = $proceed($$); }",
+                                NeowEventPatchClass));
```

Comment:
I'm not totally understanding this. How does this differ from the above patch? Should this be replacing it?

Also, if we're patching the same method, we can simply use a single static method. In this case, since they're both instrumentation patches, we can even get away with the same ExprEditor object. Part of the reason I want to patch by location rather than functionality. Because recompilation becomes quicker for the user. It does make it a bit more difficult for the devs though.
```java
                public void edit(FieldAccess fa) throws CannotCompileException {
                    if (fa.getClassName().equals(NeowEvent.class.getName()) && fa.getFieldName().equals("waitingToSave")) {
                        fa.replace(String.format(
                                "{ %1$s.doAtStartOfAct(); $_ = $proceed($$); }",
                                AtStartOfActRelicSubscriber.class.getName()));
                    }
                    if (methodCall.getClassName().equals(NeowEvent.class.getName()) &&
                            (methodCall.getMethodName().equals("blessing") || methodCall.getMethodName().equals("miniBlessing"))) {
                        methodCall.replace(String.format(
                                "{ %1$s.addDeckOfTrialsCardsIfNecessary(); $_ = $proceed($$); }",
                                NeowEventPatchClass));
                    }
                }
```